addon/internal-session.js

```diff
@@ -110,7 +128,7 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
     if (!Ember.isEmpty(this.authenticator)) {
       Ember.set(data, 'authenticated', Ember.merge({ authenticator: this.authenticator }, data.authenticated || {}));
     }
-    this.store.persist(data);
+    return this._callStoreAsync('persist', data);
   },
```

Comment:
`setUnknownProperty` still returns the value of it's `_super` so that should be ok. `_updateStore` returns a promise but is necessary to be able to wait for it for example in `_setup`